worker/src/main/scala/org/apache/celeborn/service/deploy/worker/FetchHandler.scala

```diff
@@ -204,9 +204,9 @@ class FetchHandler(
       callback: RpcResponseCallback): Unit = {
     workerSource.startTimer(WorkerSource.OPEN_STREAM_TIME, shuffleKey)
     try {
-      var fileInfo = getRawFileInfo(shuffleKey, fileName)
-      fileInfo.getPartitionType match {
-        case PartitionType.REDUCE =>
+      var fileInfo = getRawNonMemoryFileInfo(shuffleKey, fileName)
+      fileInfo.getFileMeta match {
+        case _: ReduceFileMeta =>
```

Comment:
Better to change to `meta: ReduceFileMeta` to use directly inside the branch instead of repeatedly calling `fileInfo.getFileMeta.asInstanceOf[ReduceFileMeta]`